### ISSUE TITLE
JavadocLeadingAsteriskAlign: Add indentation property

### ISSUE THREAD

**Issue by gianmarcoschifone** [checkstyle/checkstyle#20970]
After discussion in https://github.com/checkstyle/checkstyle/issues/19808

## Description
[JavadocLeadingAsteriskAlign](https://checkstyle.org/checks/javadoc/javadocleadingasteriskalign.html) ensures that leading asterisks are aligned vertically under the first asterisk (`*`) of the opening Javadoc tag.

This matches the current style:

```java
/**
 * Javadoc text.
 */
class Example { }
```

However, the examples in the Javadoc guide also show a left-side alignment style, where leading asterisks are aligned vertically under the slash (`/`) of the opening `/**` delimiter:

```java
/**
* Javadoc text.
*/
class Example { }
```

The current check cannot be configured to allow this style.

## Proposal
Add a new property `indentation` of type `int`, with default value `2`.

The property defines the expected indentation of each leading asterisk relative to the vertical position of the slash (`/`) in the opening `/**` delimiter.

- `indentation = 2` keeps the current behavior: leading asterisks are aligned under the first asterisk of `/**`.
- `indentation = 1` supports left-side alignment: leading asterisks are aligned
  under the slash of `/**`.

The same indentation rule should apply to the closing Javadoc delimiter when the check validates its alignment.

### Default values:
- `indentation`: `2`

## Config 1: Current behavior
```xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <module name="TreeWalker">
        <module name="JavadocLeadingAsteriskAlign">
            <property name="indentation" value="2"/>
        </module>
    </module>
</module>
```

### Example
```java
/**
 * Valid current style.
 */
class ValidCurrentStyle { }

/**
* Left-side alignment is invalid when indentation is 2. // violation
 */
class InvalidCurrentStyle { }
```

### Output
```
Starting audit...
[ERROR] Test.java:/:/: Leading asterisk has incorrect indentation level 1, expected is 2. [JavadocLeadingAsteriskAlign]
Audit done.
Checkstyle ends with 1 errors.
```

## Config 2: Left-side alignment
```xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <module name="TreeWalker">
        <module name="JavadocLeadingAsteriskAlign">
            <property name="indentation" value="1"/>
        </module>
    </module>
</module>
```

### Example
```java
/**
* Valid left-side alignment.
*/
class ValidLeftSideAlignment { }

/**
 * Current style is invalid when indentation is 1. // violation
*/
class InvalidLeftSideAlignment { }
```

### Output
```
Starting audit...
[ERROR] Test.java:/:/: Leading asterisk has incorrect indentation level 2, expected is 1. [JavadocLeadingAsteriskAlign]
Audit done.
Checkstyle ends with 1 errors.
```

**Comment by romani** [comment]
@gianmarcoschifone , please quote style guide sentence that requires this customization
I can not find.

**Comment by gianmarcoschifone** [comment]
@romani there is no style guide sentence actually. Customization comes from discussion with @Zopsss in:
https://github.com/checkstyle/checkstyle/issues/19808#issuecomment-4447688417
Please see also comments below that

**Comment by romani** [comment]
Doc style Requirements were from sun style.
Very ancient style guide. Let's not do this for now. We have enough to be done for openjdk , that was explicitly requested.

**Comment by romani** [comment]
`<property name="alignment" value="left"/>`

Is complicated to understand without documentation and examples.

I think we need `indentation` that will be amount of spaces from vertical  position `/` of `/*`.

`<property name="indentation" value="1"/>` to match current behavior of Check.

**Comment by gianmarcoschifone** [comment]
> `<property name="indentation" value="1"/>` to match current behavior of Check.

@romani with current check:

```
gianmarco@Mac ~/Documents/Projects/testing-checkstyle % cat Test.java
/**
* Javadoc.
 */
public class Test {}

gianmarco@Mac ~/Documents/Projects/testing-checkstyle % java -jar checkstyle-13.9.0-SNAPSHOT-all.jar -c config.xml Test.java
Starting audit...
[ERROR] /Users/gianmarco/Documents/Projects/testing-checkstyle/Test.java:2:1: Leading asterisk has incorrect indentation level 1, expected is 2. [JavadocLeadingAsteriskAlign]
Audit done.
Checkstyle ends with 1 errors.
```

"**Leading asterisk has incorrect indentation level 1, expected is 2. [JavadocLeadingAsteriskAlign]**" wouldn't this be misleading together with `<property name="indentation" value="1"/>`?

**Comment by romani** [comment]
we will change indentation values to be more logical, 
main point is that user better understand +1 +2 spaces rather than left/right.

**Comment by aclfe** [comment]
Proposed resolution to close the numbering ambiguity:

1) indentation (int, default 1, min 0) = columns the leading asterisk sits to the right of the / in /. expectedColumn = slashColumn + indentation. 1 = today's behavior (under the first *), 0 = under the slash (left alignment).
2) The current message reports the absolute column of the asterisk (that's why top-level says "expected is 2" and nested says "expected is 4/6"). I'll change the reported values to be the offset from the /, keeping the wording
"indentation level {0}, expected is {1}". Then the number equals indentation in the normal case: gianmarco's example becomes level 0, expected is 1 under the default.
3) Default 1 preserves all existing behavior; only the two logged numbers change from column to offset.

If that numbering is agreed, I'll take it. 

@romani @gianmarcoschifone

**Comment by romani** [comment]
I fully agree with your wording.

**Comment by aclfe** [comment]
Awesome. Is this approved to get started? I'm working on the pr

**Comment by gianmarcoschifone** [comment]
@aclfe I agree

**Comment by aclfe** [comment]
I am on it
